lib/elixir/lib/io/ansi/docs.ex

```diff
@@ -53,6 +55,37 @@ defmodule IO.ANSI.Docs do
     newline_after_block()
   end
 
+  @doc """
+  Prints documentation metadata (only `since` and `deprecated` for now).
+
+  See `default_options/0` for docs on the supported options.
+  """
+  @spec print_metadata(map, keyword) :: :ok
+  def print_metadata(metadata, options \\ []) when is_map(metadata) do
+    options = Keyword.merge(default_options(), options)
+    print_each_metadata(metadata, options) && newline_after_block()
+  end
+
+  defp print_each_metadata(metadata, options) do
```

Comment:
I was wondering if we might implement a more generic version. This would be my take on this code:

```elixir
@printed_metadata [:since, :deprecated]

defp print_each_metadata(metadata, options) do
  Enum.reduce(metadata, false, fn
    {key, value}, _printed when key in @printed_metadata ->
      label = color(:doc_metadata, options) <> "#{key}:" <> IO.ANSI.reset()
      indent = String.duplicate(" ", length_without_escape(label) + 1)
      write_with_wrap([label | String.split(value, @spaces)], options[:width], indent, true)

    _metadata, printed ->
      printed
  end)
end
```

What I like about this proposal is that if we decide to include new keys, developers would only need to add the key to the module attribute (instead of generating a new clause for `print_single_metadata` which I think it would be the tendency looking at the code); if we decide to print all metadata it would be just removing the second clause and the guard of the first clause; and if we decide to blacklist the keys (for example `defaults`) then we can't simply turn `when key in @printed_metadata` to `when key in @blacklisted_metadata_keys`.

Thoughts?

Btw, I don't like the naming of the module attribute 😞, we should probably find a better name.

Reply:
@fertapric I started out with that approach, but abandoned it because it's not necessary for now to be generic and we don't know what values we're going to need to deal with in the future.

For example, your snippet assumes that `value` is a string. But in the general case that will not be true. In addition we won't necessarily need or want to wrap values for other keys and we don't need to do it for `since` either.

Reply:
`when is_binary(value) and key in @metadata_keys`? 😅 

I see what you mean. I guess we can make it more generic later when we add more keys.

Reply:
@fertapric implementing your suggestion after the discussion below. Wdyt about `@metadata_filter` for the attribute name?

Reply:
`@visible_metadata`, `@printable_metadata`, `@doc_metadata_keys`? I think `@doc_metadata_keys` is the one I like the most :)

Reply:
@fertapric sorry, I didn't receive notification of this until after the commit. In any case I would either stay with `@metadata_filter` or use `@visible_metadata`.

Reply:
🤔 

`@metadata_filter` can be interpreted as a whitelist or as a blacklist. For example, `@metadata_filter [:defaults]` (blacklist) and `@metadata_filter [:since, :deprecated]` (whitelist).

So, I would go with `@visible_metadata` :)

Reply:
In Elixir `filter` typically means holding the data (i.e. Enum.filter) so I think the current semantics are fine.